src/DotVVM.Framework/Resources/Scripts/DotVVM.PostBackHandlers.ts

```diff
@@ -15,8 +25,21 @@ class ConfirmPostBackHandler extends DotvvmPostBackHandler {
     }
 }
 
-class DotvvmPostBackHandlers {
-    public confirm = (options: any) => new ConfirmPostBackHandler(<string>options.message);
+class PostbackOptions {
+    constructor(public readonly postbackId: number, public readonly sender?: HTMLElement, public readonly args : any[] = [], public readonly viewModel?: any, public readonly viewModelName?: string, public readonly validationTargetPath?: any) {}
+}
+
+class ConfirmPostBackHandler2 implements DotvvmPostbackHandler2 {
```

Comment:
I thought that you wanted to extend the interface using `interface DotvvmPostbackHandler2 { ... more props... }`, that would "work" with interfaces and don't with type. Extending using `extends` works fine, your example actually works. And you could also do `type MyPostbackHandler = DotvvmPostbackHander & { name: string }` ;)